public/Students/Leo Jayachandran/DynamicIndex/dynamicIndexMaker.js

```diff
@@ -165,7 +165,27 @@
 					break;
 				}
 			}
+			if (canPush) {
+				let startsWithLetter = false;
+				//Create list of letters
+				let alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
+				//Check if the name starts with any of the letters
+				for (var k = 0; k < alphabet.length; k++) {
+					// if doesn't start with a previous letter
+					if (!(startsWithLetter)) {
+						//Check if it starts with this letter
+						startsWithLetter = newTerm.name.toUpperCase().startsWith(alphabet[k]);
+						continue;
+					}
+					//Stop checking
+					break;
+				}
+				canPush = startsWithLetter;
```

Comment:
If desired, you can start the sorting into different letters here since you are already doing the work. You don't have to replace this since the performance impact is unnoticable.

This if statement replicates some of the work done on lines 264-280.